src/js/structs/DSLExpression.js

```diff
@@ -0,0 +1,63 @@
+import SearchDSL from '../../resources/grammar/SearchDSL.jison';
+
+/**
+ * This is a high-level compnent that wraps
+ */
+class DSLExpression {
+
+  /**
+   * Construct an immutable DSL expression
+   *
+   * @param {String} value - The expression string
+   */
+  constructor(value='') {
+    let ast = null;
+    let filter = (filters, resultset) => resultset;
+    let hasErrors = false;
+
+    if (value) {
+      // Try to parse the expression
+      try {
+        let result = SearchDSL.parse(value);
+        if (result.ast && result.filter) {
```

Comment:
Yes, because the parse function created by the JISON is throwing exceptions on errors :)
However there are cases that a result might also be produced, but be empty.